### PR TITLE
Interaction tests

### PR DESCRIPTION
**Pull request by sharwell**
Add unit tests covering additional edge cases in the overlap between different analyzers and the settings file.

### CODE REVIEW

File: StyleCop.Analyzers/StyleCop.Analyzers.Test/DocumentationRules/NoXmlFileHeaderUnitTests.cs [hunk 1]
```diff
@@ -32,6 +32,23 @@ public class NoXmlFileHeaderUnitTests : DiagnosticVerifier
 ";
 
         /// <summary>
+        /// Verifies that the analyzer will report the correct diagnostics (none for the default case) for a file
+        /// without a header.
```
Comment (pdelvo) [inline StyleCop.Analyzers/StyleCop.Analyzers.Test/DocumentationRules/NoXmlFileHeaderUnitTests.cs]: I dont understand this comment. We are expecting SA1633 here?

Comment (sharwell) [inline StyleCop.Analyzers/StyleCop.Analyzers.Test/DocumentationRules/NoXmlFileHeaderUnitTests.cs]: It was copied and pasted. I rewrote the summary for clarity here and in the original location. :+1:

**Comment by pdelvo** [comment]
:+1:
